tests/virt_autotest/libvirt_routed_virtual_network.pm

```diff
@@ -0,0 +1,120 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Managing Virtual Networks with virsh included
+# the following priorities 4 types of virtual network:
+# P1 HOST bridge network
+# P2 NAT based network
+# P3 Routed network
+# P4 Isolated network
+#
+#   This test does the following
+#    - Create Routed virtual network
+#    - Confirm Routed virtual network
+#    - Destroy Routed virtual network
+#
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "virt_feature_test_base";
+use virt_utils;
+use set_config_as_glue;
+use virtual_network_utils;
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+sub run_test {
+    my ($self) = @_;
+##P3:ROUTED NETWORK
+    #Download libvirt routed virtual network configuration files
+    my $vnet_routed_cfg_name= "vnet_routed.xml";
+    download_network_cfg($vnet_routed_cfg_name);
+
+    my $vnet_routed_clone_cfg_name= "vnet_routed_clone.xml";
+    download_network_cfg($vnet_routed_clone_cfg_name);
+
+    #Create ROUTED NETWORK
+    assert_script_run("virsh net-create vnet_routed.xml");
+    upload_logs "vnet_routed.xml";
+    assert_script_run("virsh net-create vnet_routed_clone.xml");
+    upload_logs "vnet_routed_clone.xml";
+    assert_script_run("rm -rf vnet_routed.xml vnet_routed_clone.xml");
+
+    my $gi_vnet_routed;
+    my $gi_vnet_routed_clone;
+    foreach my $guest (keys %xen::guests) {
+        record_info "$guest", "ROUTED NETWORK for $guest";
+        assert_script_run("virsh dumpxml $guest > $guest.clone");
+        assert_script_run("virsh destroy $guest");
+        assert_script_run("virsh undefine $guest");
+        assert_script_run("virsh define $guest.clone");
+        assert_script_run("rm -rf $guest.clone");
+        assert_script_run("virt-clone -o $guest -n $guest.clone -f /var/lib/libvirt/images/$guest.clone");
+        assert_script_run("virsh start $guest");
+        assert_script_run("virsh start $guest.clone");
+        assert_script_run("virsh attach-interface $guest network vnet_routed --live");
+        assert_script_run("virsh attach-interface $guest.clone network vnet_routed_clone --live");
+        #Get the Guest IP Address from ROUTED NETWORK
+        if (get_var("XEN") || check_var("HOST_HYPERVISOR", "xen")) {
+            my $mac_routed = script_output("virsh domiflist $guest | grep vnet_routed | grep -oE \"[[:xdigit:]]{2}(:[[:xdigit:]]{2}){5}\"");
+            script_retry "arp | grep $mac_routed | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"", delay => 60, retry => 6, timeout => 60;
+            $gi_vnet_routed = script_output("arp | grep $mac_routed | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
```

Comment:
I don't understand what is this variable for.

Reply:
This variable worked for get the guest ip address: $guest (guest name) -> $mac_routed (guest mac) -> $gi_vnet_routed (guest ip address from libvirt routed virtual network)

Reply:
OK, now I understand. You can make this thread as resolved.